llvm.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+################################################################################
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+################################################################################
+#
+# This script will install the llvm toolchain on the different 
+# Debian and Ubuntu versions
+
+set -eux
+
+# read optional command line argument
+LLVM_VERSION=9
+if [ "$#" -eq 1 ]; then
+    LLVM_VERSION=$1
+fi
+
+DISTRO=$(lsb_release -a | grep "Distributor ID" | awk '{print $3}')
+VERSION=$(lsb_release -a | grep Release | awk '{print $2}')
+DIST_VERSION="${DISTRO}_${VERSION}"
+
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root!" 
+   exit 1
+fi
+
+declare -A LLVM_VERSION_PATTERNS
+LLVM_VERSION_PATTERNS[8]="-8"
+LLVM_VERSION_PATTERNS[9]="-9"
+LLVM_VERSION_PATTERNS[10]=""
+
+if [ ! ${LLVM_VERSION_PATTERNS[$LLVM_VERSION]+_} ]; then
+    echo "This script does not support LLVM version $LLVM_VERSION"
+    exit 3
+fi
+
+LLVM_VERSION_STRING=${LLVM_VERSION_PATTERNS[$LLVM_VERSION]}
+
+# find the right add-apt-repository arguments for the distros
+declare -A APT_REPOS
+APT_REPOS["Debian_9.9"]="deb http://apt.llvm.org/stretch/ llvm-toolchain-stretch$LLVM_VERSION_STRING main"
```

Comment:
I guess it won't work if I use Debian 9.8 ?

Reply:
No, but I can add 9.8 if you want. 
Not sure if I can find a docker image for testing though...

Reply:
my point was more "how do we make this scale without human intervention?" :)

Reply:
Gotcha. I could replace the hash table with a switch/case and use pattern matching:

```bash
case $DIST_VERSION in
"Debian_9.*" ) 
    APT_REPO=...
   ;;
"Debian_10.*" ) APT_REPO=...
    APT_REPO=...
   ;;
...
```

At least stackoverflow says that this should work, I did not test it though...

Reply:
yeah, probably better

Reply:
@ChristianKuehnel Do you want me to land this version or you will fix it before? thanks

Reply:
@sylvestre I just uploaded the changed script.